src/ORM/DataObject.php

```diff
@@ -2485,6 +2489,65 @@ public function scaffoldFormFields($_params = null)
         return $fs->getFieldList();
     }
 
+    /**
+     * Scaffold a form field for selecting records of this model type in a has_one relation.
+     */
+    public function scaffoldFormFieldForHasOne(
+        string $fieldName,
+        ?string $fieldTitle,
+        string $relationName,
+        DataObject $ownerRecord
+    ): FormField {
+        $labelField = $this->hasField('Title') ? 'Title' : 'Name';
```

Comment:
What happens if the DataObject lacks both a 'Title' and a 'Name' field? Does it need to fallback to an 'ID' column so that there's at least not an exception?

Reply:
I copied this from the existing logic in `DBForeignKey`
```php
$labelField = $hasOneSingleton->hasField('Title') ? 'Title' : 'Name';
```
Changing how the label is figured out isn't in scope for this PR.